Rock.Messaging.RabbitMQ/RabbitEngine.cs

```diff
@@ -0,0 +1,110 @@
+                HostName = connectionStrs[0],
+                Port = int.Parse(connectionStrs[1]),
+                VirtualHost = config.vHost,
+            };
+            connection = connectionFactory.CreateConnection();
+            
+        }
+
+        public void Dispose()
+        {
+            connection.Dispose();
+            foreach (var m in consumerModels)
+            {
+                m.Abort();
+            }
+            
+        }
+
+        #region Receiver
+        public void Start(string selector) // TODO: Discuss: RabbitMQ supports getting individual messages- should Rock.Messaging support that?
+        {
+            var model = connection.CreateModel();
+            model.ExchangeDeclare(config.Exchange, config.ExchangeType ?? (topical ? "topic": "direct"), true); // Should perhaps configure this default somehow? Also RabbitMQ Topics might not fulfill the same intent as the abstraction?
+            model.QueueDeclare(config.QueueName, true, false, false, null); // TODO: Discuss: Add config settings for these? It's already pretty cluttered...
+            if (!string.IsNullOrWhiteSpace(selector))
+            {
+                model.QueueBind(config.QueueName, config.Exchange, selector);
+            }
+            else
+            {
+                model.QueueBind(config.QueueName, config.Exchange, config.RoutingKey ?? string.Empty);
+            }
+            
+            var consumer = new EventingBasicConsumer(model);
+            consumer.Received += delegate(object sender, BasicDeliverEventArgs args)
+            {
+                var messageReceived = new MessageReceivedEventArgs(new RabbitReceiverMessage(args, model));
+                OnMessageReceived(messageReceived);
+                if (config.AutoAcknowledge)
+                {
+                    model.BasicAck(args.DeliveryTag, false);
+                }
+            };
+            model.BasicQos(0, config.MaxRequests, true);
+            model.BasicConsume(config.QueueName, false, consumer);
+        }
+
+        protected virtual void OnMessageReceived(MessageReceivedEventArgs args)
+        {
+            var handler = MessageReceived;
+
+            if (handler != null)
+            {
+                handler(this, args);
+            }
+        }
+        public event EventHandler<MessageReceivedEventArgs> MessageReceived;
+
+        #endregion
+        public Task SendAsync(ISenderMessage message)
+        {
+            return Task.Run(delegate
+            {
+                using (var model = connection.CreateModel())
+                {
+                    var props = model.CreateBasicProperties();
+                    props.Headers = message.Headers.ToDictionary<KeyValuePair<string, string>, string, object>(kvp => kvp.Key, kvp => kvp.Value); // IEnum<KVP<S,S>> in, Dictionary<string, object> out.
+                    props.ContentType = message.MessageFormat.ToString();
+                    props.Priority = message.Priority ?? 0;
```

Comment:
Also, probably need some validation around `Priority` values. What happens if a value of 15 or 255 is used?